docker-app/qfieldcloud/core/tests/test_jobs.py

```diff
@@ -0,0 +1,173 @@
+import logging
+
+from qfieldcloud.authentication.models import AuthToken
+from qfieldcloud.core.models import (
+    Job,
+    PackageJob,
+    Person,
+    ProcessProjectfileJob,
+    Project,
+)
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.test import APITransactionTestCase
+
+from .utils import (
+    set_subscription,
+    setup_subscription_plans,
+    testdata_path,
+    wait_for_project_ok_status,
+)
+
+logging.disable(logging.CRITICAL)
+
+VALID_LOCALIZED_LAYER_KEY = "apiary_853c29cb_ead8_4932_a60a_5cba33140b3d"
+INVALID_LAYER_KEY = "apiary_e7d1b542_1b1c_422d_ac75_c14d2b54e472"
+INVALID_LOCALIZED_LAYER_KEY = "area_a7b70aff_23f2_4876_9d4e_fa63036df0b2"
+
+
+class QfcTestCase(APITransactionTestCase):
+    def setUp(self):
+        setup_subscription_plans()
+
+        # Create a user
+        self.u1 = Person.objects.create_user(username="u1", password="abc123")
+        self.t1 = AuthToken.objects.get_or_create(user=self.u1)[0]
+
+        # Create a project
+        self.p1 = Project.objects.create(name="p1", is_public=False, owner=self.u1)
+
+    def upload_file(self, project_id: str, local: str, remote: str) -> Response:
```

Comment:
```suggestion
    def upload_file(
  	    self,
  	    project: Project,
  	    local_filename: str,
  	    remote_filename: str,
  	) -> Response:
```

Better pass the whole project object. Not that your typing was wrong, since you set the `project_id: str`, but `isinstance(project.id, UUID)`, you have to explicitly convert to `str`. Check if you can configure your editor to check this for you.

As for local/remote, it is not clear what they are, avoid using adjectives as variable names, add a noun whenever possible.

Last comment, I would make the `remote_filename` optional and use the `local_filename` value by default. Will save a few bytes of writing.